doc/en/components/grids/tree-grid/load-on-demand.md

```diff
@@ -0,0 +1,118 @@
+---
+title: {Platform} Tree Grid | Fastest {Platform} Tree Table | Infragistics
+_description: The {ProductName} Tree Grid provides the necessary tools to load child data on demand when a parent row is expanded. That way the volume of data would be greatly reduced and can be retrieved only when the user needs it.
+_keywords: {Platform} tree grid, igniteui for {Platform}, infragistics
+---
+
+# Tree Grid Load On Demand
+
+The Ignite UI for {Platform} Tree Grid can be rendered in such way that it requires the minimal amount of data to get from the server so the user could see it as quickly as possible. Then, only after the user expands a row, the children for that particular parent row will be loaded. This mechanism, also known as Load on Demand, can be easily configured to work with any remote data.
+
+## {Platform} Tree Grid Load On Demand Example
+
+`sample="/{TreeGridSample}/load-on-demand", height="700", alt="{Platform} Tree Grid Load On Demand Example"`
+
+### Usage
+
+The Load on Demand feature is compatible with both types of Tree Grid data sources - primary and foreign keys or child collection. You only need to load the root level data in the Tree Grid and specify the necessary keys for one of the data source types. In order to load the child rows when the user expands a row, the Tree Grid provides the callback input property `LoadChildrenOnDemand`.
+
+<!-- Angular -->
+
+```html
+<igx-tree-grid #treeGrid [data]="data" primaryKey="ID" foreignKey="ParentID"
+        [loadChildrenOnDemand]="loadChildren">
+</igx-tree-grid>
+```
+
+<!-- end: Angular -->
+
+```html
+<igc-tree-grid id="treeGrid"></igc-tree-grid>
+```
+
+```ts
+constructor() {
+    var treeGrid = document.getElementById("treeGrid") as IgcTreeGridComponent;
+    treeGrid.data = this.employeesFlatData;
+    treeGrid.loadChildrenOnDemand = (parentID: any, done: (children: any[]) => void) => {
+    this.getData(parentID, (children) => done(children));
+    };
+}
+```
+
+The LoadChildrenOnDemand callback provides two parameters:
+
+- parentID - the ID of the parent row that is being expanded.
+- done - callback that should be called with the children when they are retrieved from the server. 
+
+```typescript
+public loadChildren = (parentID: any, done: (children: any[]) => void) => {
+    this.getData(parentID, (children) => done(children));
+}
+```
```

Comment:
build flag lines 29-52 for WC in case this topic is later enabled for Blazor

Reply:
@MayaKirova Can you take a look?

Reply:
I don't see anything that would necessarily need to be excluded for blazor between those lines but ok.